### PR TITLE
Add SMTP modules (smtp, smtp_info) for US10.

### PR DESCRIPTION
**Pull request by anazobec**
Added SMTP modules for US10:
- module_utils: smtp
- modules: smtp, smtp_info

Example usages are:
```yaml
# ---------- Module smtp
- name: Modify SMTP configuration without authentication
  scale_computing.hypercore.smtp:
    server: smtp-relay.gmail.com
    port: 25
    use_ssl: False
    requires_auth: False
    from_address: example@example.com

- name: Modify SMTP configuration with authentication
  scale_computing.hypercore.smtp:
    server: smtp-relay.gmail.com
    port: 25
    use_ssl: False
    requires_auth: True
    username: example
    password: example123
    from_address: example@example.com

- name: Modify SMTP configuraiton with required params only
  scale_computing.hypercore.smtp:
    server: smtp-relay.gmail.com
    port: 25

# ---------- Module smtp_info
- name: List SMTP Configuration
  scale_computing.hypercore.smtp_info:
  register: smtp_info
- ansible.builtin.debug:
    msg: "{{ smtp_info }}"
```

### CODE REVIEW

File: plugins/modules/smtp.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        endpoint="{0}/{1}".format("/rest/v1/AlertSMTPConfig", smtp.uuid),
+        payload=payload,
+        check_mode=module.check_mode,
+    )
+
+    TaskTag.wait_task(rest_client, update_task_tag)  # wait for task to finish
+
+    record = SMTP.get_state(rest_client)
+    diff = dict(before=old_state, after=record)
+    change = old_state != record
```
Comment (justinc1) [inline plugins/modules/smtp.py:226]: Here you compute `change` a second time. Why? You can also assert that `old_state != record` is True.

File: plugins/modules/smtp.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                type="str",
+                default="",
+                no_log=True,
+                required=False,
+            ),
+            from_address=dict(
+                type="str",
+                default="",
+                required=False,
+            ),
```
Comment (justinc1) [inline plugins/modules/smtp.py:273]: `from_address` is optional in GUI? It is, this is OK.

File: plugins/modules/smtp.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+      modified: 1675435601
+      nodeUUIDs: []
+      objectUUID: smtpconfig_guid
+      progressPercent: 100
+      sessionID: 92b4a736-259c-4f3c-9492-ce0c36691372
+      state: COMPLETE
+      taskTag: 761
+    port: 25
+    smtp_server: smtp-relay.gmail.com
+    use_auth: false
```
Comment (justinc1) [inline plugins/modules/smtp.py:115]: I think we should return also `username`, if we get it from API.

File: plugins/modules/smtp_info.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+"""
+
+
+EXAMPLES = r"""
+- name: List all configurations on DNS configuration on HyperCore API
+  scale_computing.hypercore.smtp_info:
+  register: smtp_info
+"""
+
+RETURN = r"""
```
Comment (justinc1) [inline plugins/modules/smtp_info.py:34]: Please return also `username` if we get it from API.

Comment (justinc1) [reply plugins/modules/smtp_info.py:34]: auth_user - it is there, document it.

File: plugins/modules/smtp.py (hunk 3 above)
Comment (justinc1) [reply plugins/modules/smtp.py:115]: auth_user - it is there, document it.

File: tests/integration/targets/smtp/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        default_use_ssl: "{{ smtp_config.record.0.useSSL }}"
+        default_use_auth: "{{ smtp_config.record.0.useAuth }}"
+        default_auth_user: "{{ smtp_config.record.0.authUser }}"
+        default_auth_password: "{{ smtp_config.record.0.authPassword }}"
+        default_from_address: "{{ smtp_config.record.0.fromAddress }}"
+        default_uuid: "{{ smtp_config.record.0.uuid }}"
+
+    # -----------------------------------------------------------
+
+#    - name: Modify SMTP configs - REQUIRED ONLY
```
Comment (justinc1) [inline tests/integration/targets/smtp/tasks/main.yml:34]: unfinished, we can discuss what to do if needed

**Review by justinc1 — CHANGES_REQUESTED** [review]
Only minor changes

File: changelogs/fragments/SMTP_modules.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (justinc1) [inline changelogs/fragments/SMTP_modules.yml:1]: smtp_modules.yml - we have all filename in lowercase

File: plugins/modules/smtp.py [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+    use_ssl: false
+    uuid: smtpconfig_guid
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from typing import Tuple
```
[the remark this answers is not in the document]
Comment (anazobec) [reply plugins/modules/smtp.py:122]: Empty string.

File: examples/smtp.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    requires_auth: false
+    from_address: example@example.com
+
+- name: Modify SMTP configuration with authentication
+  scale_computing.hypercore.smtp:
+    server: smtp-relay.gmail.com
+    port: 25
+    use_ssl: false
+    requires_auth: true
+    username: example
```
Comment (justinc1) [inline examples/smtp.yml:16]: auth_user, auth_pasword

**Pull request merged**